internal/fnapi/userauth.go

```diff
@@ -29,15 +30,27 @@ type UserAuth struct {
 }
 
 func (user UserAuth) GenerateToken(ctx context.Context) (string, error) {
-	if user.Clerk != nil {
+	switch {
+	case user.Clerk != nil:
 		jwt, err := clerk.JWT(ctx, user.Clerk)
 		if err != nil {
 			return "", err
 		}
 		return fmt.Sprintf("jwt:%s", jwt), nil
-	}
 
-	return base64.RawStdEncoding.EncodeToString(user.InternalOpaque), nil
+	case user.InternalOpaque != nil:
+		return base64.RawStdEncoding.EncodeToString(user.InternalOpaque), nil
+
+	case os.Getenv("GITHUB_ACTIONS") == "true":
+		jwt, err := github.JWT(ctx, "") // XXX: do not set "audience" and use default one
+		if err != nil {
+			return "", err
+		}
+		return fmt.Sprintf("gh-jwt:%s", jwt), nil
+
+	default:
+		return "", fnerrors.New("no authentication method available")
```

Comment:
We can actually return auth.ErrRelogin here, since ns login will populate user.InternalOpaque and hence avoid this error.

Reply:
done